src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/FeatureSwitchManager.cs

```diff
@@ -641,19 +688,68 @@ private bool TryGetConstantArgument(MethodIL methodIL, byte[] body, OpcodeFlags[
             return false;
         }
 
+        private static string GetResourceStringForAccessor(EcmaMethod method)
+        {
+            Debug.Assert(method.Name.StartsWith("get_"));
+            string resourceStringName = method.Name.Substring(4);
+
+            EcmaModule module = method.Module;
+
+            string resourceName1 = $"{module.Assembly.GetName().Name}.Strings.resources";
+            string resourceName2 = $"FxResources.{module.Assembly.GetName().Name}.SR.resources";
```

Comment:
We now need to inline knowledge about these names. Don't like this one either. Don't see way out.

Reply:
How does resource manager find it? In theory this has to be written somewhere in the compiled code, so the compiler could get it from there. But it's probably VERY complicated to do so.

Reply:
Resource manager grabs it from a piece of (generated) code. It looks like this:

```csharp
namespace FxResources.System.Private.DisabledReflection
{
    internal static class SR { }
}
namespace System
{
    internal static partial class SR
    {
        private static global::System.Resources.ResourceManager s_resourceManager;
        internal static global::System.Resources.ResourceManager ResourceManager => s_resourceManager ?? (s_resourceManager = new global::System.Resources.ResourceManager(typeof(FxResources.System.Private.DisabledReflection.SR)));
    }
}
```

So we'd need to crack open the cctor on the System.SR class and find the type that is passed to resource manager's constructor. That's the name of the resource.

It's not impossible. But also not completely straightforward.

Reply:
Hm, but this is all generated by Arcade logic, so we can just change it: https://github.com/dotnet/arcade/blob/083fc0406279952794c853630a08a141eef7244f/src/Microsoft.DotNet.Arcade.Sdk/src/GenerateResxSource.cs

Looks like the extra type is only done to comfort .NET Native (likely also some old version of .NET Native that still did ILMerging base on the comment).

So we can remove the useless type and instead call into the ResourceManager ctor that takes the resource name.

That will be a net improvement everywhere. I like those kinds of changes.

```
namespace System
{
    internal static partial class SR
    {
        private const string ResourceName = "FxResources.System.Private.DisabledReflection.SR";
        private static global::System.Resources.ResourceManager s_resourceManager;
        internal static global::System.Resources.ResourceManager ResourceManager => s_resourceManager ?? (s_resourceManager = new global::System.Resources.ResourceManager(typeof(SR).Assembly), ResourceName);
    }
}
```

Reply:
Nice - although in probably won't help that much to detect it from the compiler. We would still have to understand the .cctor and alike. I'd say keep the current "hardcoded" version and file a bug that this could be improved by analyzing the static .cctor. Even that would still rely on just specific patterns, but it would more robust.

Reply:
We'd be looking for the const string - that one is easy to find and can be just part of the protocol.